src/plugin_proc.h

```diff
@@ -1,31 +1,232 @@
 #ifndef NETDATA_PLUGIN_PROC_H
 #define NETDATA_PLUGIN_PROC_H 1
 
+/**
+ * @file plugin_proc.h
+ * @brief Thread to collect linux proc filesystem metrics.
+ *
+ * __How to add a data collector__
+ * \todo Describe how to add a data collector.
```

Comment:
I will leave this todo there if you are OK with it. This is a time consuming work and I do not see the need to have it in this pull request.